File: src/test/java/xyz/javecs/tools/expr/CalculatorTest.java
package xyz.javecs.tools.expr;

import org.junit.Assert;
import org.junit.Test;

public class CalculatorTest {
    @Test
    public void testCalculator() {
        Calculator calc = new Calculator();
        calc.eval("3 + 4");
        Number value = calc.getValue();
        Assert.assertEquals(7, value);
    }
}

File: src/test/kotlin/xyz/javecs/tools/expr/CalculatorTest.kt
package xyz.javecs.tools.expr

import kotlin.test.assertEquals
import org.junit.Test

class CalculatorTest {
    @Test fun calcExpr1() {
        val expr = "3 + 4"
        assertEquals(7, Calculator(arrayOf(expr)).evaluate().getValue())
    }

    @Test fun calcExpr2() {
        val expr = "3 + 4"
        assertEquals(7, Calculator().evaluate(expr).getValue())
    }

    @Test fun calcExpr3() {
        val calc = Calculator(arrayOf("3 + 4", "30 + 40"))
        assertEquals(70, calc.evaluate().getValue())
    }

    @Test fun calcExpr4() {
        val calc = Calculator()
        assertEquals(7, calc.evaluate("3 + 4").getValue())
        assertEquals(8, calc.evaluate("3 + 5").getValue())
    }

    @Test fun calcExpr5() {
        val calc = Calculator(arrayOf("30 + 40"))
        assertEquals("70", calc.evaluate().toString())
    }
}
